lib/rubygems/specification.rb

```diff
@@ -2822,7 +2822,7 @@ def validate_dependencies # :nodoc:
       end
 
       warning_messages << "prerelease dependency on #{dep} is not recommended" if
-        prerelease_dep
+        prerelease_dep && !version.prerelease?
```

Comment:
Please move this to a separate pull request as it is separate from disallowing warnings.